build.gradle.kts

```diff
@@ -0,0 +1,73 @@
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+import org.gradle.api.tasks.testing.logging.TestLogEvent
+
+plugins {
+    id("java-gradle-plugin")
+    id("com.gradle.plugin-publish") version "0.12.0"
+    groovy //TODO remove it at the end of the migration
+    id("com.adarshr.test-logger") version "2.1.1" //TODO latest version incompatible with existing groovy version
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+repositories {
+    mavenCentral()
+    gradlePluginPortal()
+}
+
+dependencies {
+    implementation("io.gatling:gatling-enterprise-plugin-commons:1.5.5") {
+        exclude("com.fasterxml.jackson.core")
+    }
+    implementation("com.fasterxml.jackson.core:jackson-databind:2.14.3")
```

Comment:
why downgrading?

Reply:
2.15.2 seems to be incompatible with this env: 

```
Caused by: org.gradle.api.GradleException: Failed to create Jar file /private/var/folders/5h/m8140vp91ps15nvjscvffhdm0000gn/T/.gradle-test-kit-christophe/caches/jars-8/61ec2da8f88e12427195250c3254cf41/jackson-core-2.15.2.jar.
	at org.gradle.internal.classpath.ClasspathBuilder.jar(ClasspathBuilder.java:47)
	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.instrument(InstrumentingClasspathFileTransformer.java:103)
	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.transform(InstrumentingClasspathFileTransformer.java:98)
	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.transform(InstrumentingClasspathFileTransformer.java:68)
	at org.gradle.internal.classpath.DefaultCachedClasspathTransformer$TransformFile.lambda$schedule$0(DefaultCachedClasspathTransformer.java:261)
	... 3 more
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 61
	at org.objectweb.asm.ClassReader.<init>(ClassReader.java:196)
```

Reply:
WTF
Can you provide the full stacktrace please?

Reply:
```
BUILD FAILED in 91ms
[0K[1mfunc.WhenRunFailingSimulationSpec[22m should execute all simulations even if one fails because of gatling runtime[31m FAILED[31m
  Condition failed with Exception:
  
  ex.buildResult.task(":$GATLING_RUN_TASK_NAME").outcome == FAILED
  |  |           |       |                       |
  |  |           null    gatlingRun              java.lang.NullPointerException: Cannot get property 'outcome' on null object
  |  |                                           	at func.WhenRunFailingSimulationSpec.should execute all simulations even if one fails because of gatling runtime(WhenRunFailingSimulationSpec.groovy:112)
  |  <org.gradle.testkit.runner.internal.FeatureCheckBuildResult@5552e321 delegateBuildResult=org.gradle.testkit.runner.internal.DefaultBuildResult@1256c018 outputFeatureCheck=org.gradle.testkit.runner.internal.feature.BuildResultOutputFeatureCheck@3eb6e36d>
  org.gradle.testkit.runner.UnexpectedBuildFailure: Unexpected build execution failure in /var/folders/5h/m8140vp91ps15nvjscvffhdm0000gn/T/junit3115439502231092236 with arguments [--stacktrace, -Dgatling.hostName=HTTP://COMPUTER-DATABASE.GATLING.IO, gatlingRun]
   
  Output:
   
  FAILURE: Build failed with an exception.
   
  * What went wrong:
  Gradle could not start your build.
  > Could not create service of type ScriptPluginFactory using BuildScopeServices.createScriptPluginFactory().
     > Could not create service of type ClientInjectedClasspathPluginResolver using BuildScopeServices.createInjectedClassPathPluginResolver().
        > Failed to create Jar file /private/var/folders/5h/m8140vp91ps15nvjscvffhdm0000gn/T/.gradle-test-kit-christophe/caches/jars-8/61ec2da8f88e12427195250c3254cf41/jackson-core-2.15.2.jar.
   
  * Try:
  Run with --info or --debug option to get more log output. Run with --scan to get full insights.
   
  * Exception is:
  org.gradle.initialization.exception.InitializationException: Gradle could not start your build.
  	at org.gradle.initialization.exception.DefaultExceptionAnalyser.collectFailures(DefaultExceptionAnalyser.java:65)
  	at org.gradle.initialization.exception.MultipleBuildFailuresExceptionAnalyser.transform(MultipleBuildFailuresExceptionAnalyser.java:48)
  	at org.gradle.initialization.exception.StackTraceSanitizingExceptionAnalyser.transform(StackTraceSanitizingExceptionAnalyser.java:30)
  	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:64)
  	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:41)
  	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:47)
  	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:31)
  	at org.gradle.tooling.internal.provider.StdInSwapExecuter$1.create(StdInSwapExecuter.java:43)
  	at org.gradle.tooling.internal.provider.StdInSwapExecuter$1.create(StdInSwapExecuter.java:40)
  	at org.gradle.util.Swapper.swap(Swapper.java:38)
  	at org.gradle.tooling.internal.provider.StdInSwapExecuter.execute(StdInSwapExecuter.java:40)
  	at org.gradle.tooling.internal.provider.StdInSwapExecuter.execute(StdInSwapExecuter.java:29)
  	at org.gradle.tooling.internal.provider.DaemonBuildActionExecuter.execute(DaemonBuildActionExecuter.java:46)
  	at org.gradle.tooling.internal.provider.DaemonBuildActionExecuter.execute(DaemonBuildActionExecuter.java:31)
  	at org.gradle.tooling.internal.provider.LoggingBridgingBuildActionExecuter.execute(LoggingBridgingBuildActionExecuter.java:60)
  	at org.gradle.tooling.internal.provider.LoggingBridgingBuildActionExecuter.execute(LoggingBridgingBuildActionExecuter.java:37)
  	at org.gradle.tooling.internal.provider.ProviderConnection.run(ProviderConnection.java:218)
  	at org.gradle.tooling.internal.provider.ProviderConnection.run(ProviderConnection.java:144)
  	at org.gradle.tooling.internal.provider.DefaultConnection.getModel(DefaultConnection.java:202)
  	at org.gradle.tooling.internal.consumer.connection.CancellableModelBuilderBackedModelProducer.produceModel(CancellableModelBuilderBackedModelProducer.java:54)
  	at org.gradle.tooling.internal.consumer.connection.PluginClasspathInjectionSupportedCheckModelProducer.produceModel(PluginClasspathInjectionSupportedCheckModelProducer.java:38)
  	at org.gradle.tooling.internal.consumer.connection.AbstractConsumerConnection.run(AbstractConsumerConnection.java:64)
  	at org.gradle.tooling.internal.consumer.connection.ParameterValidatingConsumerConnection.run(ParameterValidatingConsumerConnection.java:49)
  	at org.gradle.tooling.internal.consumer.DefaultBuildLauncher$1.run(DefaultBuildLauncher.java:97)
  	at org.gradle.tooling.internal.consumer.DefaultBuildLauncher$1.run(DefaultBuildLauncher.java:89)
  	at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:144)
  	at org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.run(CancellableConsumerActionExecutor.java:45)
  	at org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.run(ProgressLoggingConsumerActionExecutor.java:61)
  	at org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.run(RethrowingErrorsConsumerActionExecutor.java:38)
  	at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor$1$1.run(DefaultAsyncConsumerActionExecutor.java:67)
  	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
  	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
  	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
  Caused by: org.gradle.internal.service.ServiceCreationException: Could not create service of type ScriptPluginFactory using BuildScopeServices.createScriptPluginFactory().
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryMethodService.invokeMethod(DefaultServiceRegistry.java:848)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.create(DefaultServiceRegistry.java:773)
  	at org.gradle.internal.service.DefaultServiceRegistry$ManagedObjectServiceProvider.getInstance(DefaultServiceRegistry.java:563)
  	at org.gradle.internal.service.DefaultServiceRegistry$SingletonService.get(DefaultServiceRegistry.java:621)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.assembleParameters(DefaultServiceRegistry.java:786)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.create(DefaultServiceRegistry.java:772)
  	at org.gradle.internal.service.DefaultServiceRegistry$ManagedObjectServiceProvider.getInstance(DefaultServiceRegistry.java:563)
  	at org.gradle.internal.service.DefaultServiceRegistry$SingletonService.get(DefaultServiceRegistry.java:621)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.assembleParameters(DefaultServiceRegistry.java:786)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.create(DefaultServiceRegistry.java:772)
  	at org.gradle.internal.service.DefaultServiceRegistry$ManagedObjectServiceProvider.getInstance(DefaultServiceRegistry.java:563)
  	at org.gradle.internal.service.DefaultServiceRegistry$SingletonService.get(DefaultServiceRegistry.java:621)
  	at org.gradle.internal.service.DefaultServiceRegistry.find(DefaultServiceRegistry.java:305)
  	at org.gradle.internal.service.DefaultServiceRegistry.get(DefaultServiceRegistry.java:289)
  	at org.gradle.internal.service.DefaultServiceRegistry.get(DefaultServiceRegistry.java:284)
  	at org.gradle.initialization.DefaultGradleLauncherFactory.createDefaultGradleLauncher(DefaultGradleLauncherFactory.java:174)
  	at org.gradle.initialization.DefaultGradleLauncherFactory.doNewInstance(DefaultGradleLauncherFactory.java:158)
  	at org.gradle.initialization.DefaultGradleLauncherFactory.doNewInstance(DefaultGradleLauncherFactory.java:101)
  	at org.gradle.initialization.DefaultGradleLauncherFactory.newInstance(DefaultGradleLauncherFactory.java:80)
  	at org.gradle.composite.internal.DefaultRootBuildState.<init>(DefaultRootBuildState.java:47)
  	at org.gradle.composite.internal.DefaultIncludedBuildRegistry.createRootBuild(DefaultIncludedBuildRegistry.java:90)
  	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:57)
  	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:30)
  	at org.gradle.launcher.exec.BuildTreeScopeLifecycleBuildActionExecuter.lambda$execute$0(BuildTreeScopeLifecycleBuildActionExecuter.java:33)
  	at org.gradle.internal.buildtree.BuildTreeState.run(BuildTreeState.java:53)
  	at org.gradle.launcher.exec.BuildTreeScopeLifecycleBuildActionExecuter.execute(BuildTreeScopeLifecycleBuildActionExecuter.java:32)
  	at org.gradle.launcher.exec.BuildTreeScopeLifecycleBuildActionExecuter.execute(BuildTreeScopeLifecycleBuildActionExecuter.java:27)
  	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:104)
  	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:55)
  	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:64)
  	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:37)
  	at org.gradle.tooling.internal.provider.SessionScopeLifecycleBuildActionExecuter.lambda$execute$0(SessionScopeLifecycleBuildActionExecuter.java:54)
  	at org.gradle.internal.session.BuildSessionState.run(BuildSessionState.java:67)
  	at org.gradle.tooling.internal.provider.SessionScopeLifecycleBuildActionExecuter.execute(SessionScopeLifecycleBuildActionExecuter.java:50)
  	at org.gradle.tooling.internal.provider.SessionScopeLifecycleBuildActionExecuter.execute(SessionScopeLifecycleBuildActionExecuter.java:36)
  	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:36)
  	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:25)
  	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:59)
  	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:31)
  	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:55)
  	... 29 more
  Caused by: org.gradle.internal.service.ServiceCreationException: Could not create service of type ClientInjectedClasspathPluginResolver using BuildScopeServices.createInjectedClassPathPluginResolver().
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryMethodService.invokeMethod(DefaultServiceRegistry.java:848)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.create(DefaultServiceRegistry.java:773)
  	at org.gradle.internal.service.DefaultServiceRegistry$ManagedObjectServiceProvider.getInstance(DefaultServiceRegistry.java:563)
  	at org.gradle.internal.service.DefaultServiceRegistry$SingletonService.get(DefaultServiceRegistry.java:621)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.assembleParameters(DefaultServiceRegistry.java:786)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.create(DefaultServiceRegistry.java:772)
  	at org.gradle.internal.service.DefaultServiceRegistry$ManagedObjectServiceProvider.getInstance(DefaultServiceRegistry.java:563)
  	at org.gradle.internal.service.DefaultServiceRegistry$SingletonService.get(DefaultServiceRegistry.java:621)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.assembleParameters(DefaultServiceRegistry.java:786)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.create(DefaultServiceRegistry.java:772)
  	at org.gradle.internal.service.DefaultServiceRegistry$ManagedObjectServiceProvider.getInstance(DefaultServiceRegistry.java:563)
  	at org.gradle.internal.service.DefaultServiceRegistry$SingletonService.get(DefaultServiceRegistry.java:621)
  	at org.gradle.internal.service.DefaultServiceRegistry.find(DefaultServiceRegistry.java:305)
  	at org.gradle.internal.service.DefaultServiceRegistry.get(DefaultServiceRegistry.java:289)
  	at org.gradle.internal.service.DefaultServiceRegistry.get(DefaultServiceRegistry.java:284)
  	at org.gradle.internal.service.scopes.BuildScopeServices.defaultScriptPluginFactory(BuildScopeServices.java:409)
  	at org.gradle.internal.service.scopes.BuildScopeServices.createScriptPluginFactory(BuildScopeServices.java:396)
  	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
  	at org.gradle.internal.service.ReflectionBasedServiceMethod.invoke(ReflectionBasedServiceMethod.java:34)
  	at org.gradle.internal.service.DefaultServiceRegistry$FactoryMethodService.invokeMethod(DefaultServiceRegistry.java:846)
  	... 68 more
  Caused by: org.gradle.api.GradleException: Failed to create Jar file /private/var/folders/5h/m8140vp91ps15nvjscvffhdm0000gn/T/.gradle-test-kit-christophe/caches/jars-8/61ec2da8f88e12427195250c3254cf41/jackson-core-2.15.2.jar.
  	at org.gradle.internal.classpath.ClasspathBuilder.jar(ClasspathBuilder.java:47)
  	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.instrument(InstrumentingClasspathFileTransformer.java:103)
  	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.transform(InstrumentingClasspathFileTransformer.java:98)
  	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.transform(InstrumentingClasspathFileTransformer.java:68)
  	at org.gradle.internal.classpath.DefaultCachedClasspathTransformer$TransformFile.lambda$schedule$0(DefaultCachedClasspathTransformer.java:261)
  	... 3 more
  Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 61
  	at org.objectweb.asm.ClassReader.<init>(ClassReader.java:196)
  	at org.objectweb.asm.ClassReader.<init>(ClassReader.java:177)
  	at org.objectweb.asm.ClassReader.<init>(ClassReader.java:163)
  	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.lambda$visitEntries$1(InstrumentingClasspathFileTransformer.java:116)
  	at org.gradle.internal.classpath.ClasspathWalker.visitJarContents(ClasspathWalker.java:91)
  	at org.gradle.internal.classpath.ClasspathWalker.visit(ClasspathWalker.java:55)
  	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.visitEntries(InstrumentingClasspathFileTransformer.java:114)
  	at org.gradle.internal.classpath.InstrumentingClasspathFileTransformer.lambda$instrument$0(InstrumentingClasspathFileTransformer.java:105)
  	at org.gradle.internal.classpath.ClasspathBuilder.buildJar(ClasspathBuilder.java:58)
  	at org.gradle.internal.classpath.ClasspathBuilder.jar(ClasspathBuilder.java:45)
  	... 7 more
```

Reply:
It's a gradle bug, see: https://github.com/FasterXML/jackson-core/issues/955

Reply:
and we can't update gradle cause https://github.com/gatling/gatling-gradle-plugin/commit/b7ce993e4e954ef4df5031e925c078ba820c250e#diff-40640fe1078ece83d7ea8fb67daacd77923a86d13447baf9769660b3b46f2eceR3 ?

Reply:
We can't upgrade gradle used for building because of https://github.com/gradle/gradle/issues/18570 which is still open  and breaks gradle 6 runtime support.

gradle 6.9.4 was released on Feb 22, 2023, just a few month ago. And we lack data to be able to decide if we can afford require gradle 7 as the minimal version.

Reply:
O?, fine with the downgrade until we upgrade the minimum supported gradle version to 7.6.1.